src/game_logic/interfaces.ts

```diff
@@ -80,76 +81,58 @@ export interface Contract {
   readonly name: string
   // Long form description / flavor text
   readonly description: string
-  /**
-   * Look at the current game state and determine if this contract has been fulfilled
-   *
-   * changes `isCompeleted()` if has been completed
-   *
-   * DO NOT modify the game state
-   * @param state
-   */
-  check(state: GameState): boolean
-
-  /**
-   * Modify the current game state to apply the rewards of this contract
-   * @param state
-   */
-  reward(state: GameState): void
-
-  /**
-   * Indicates if this contract has been resolved
-   */
+  // Indicates if this contract has been resolved
   readonly completed: boolean
 }
 
 export interface MapState {
-  tiles: Array<Tile>,
-  size: {
-    x: number,
-    y: number
+  readonly tiles: ReadonlyArray<Tile>,
+  readonly size: {
+    readonly x: number,
+    readonly y: number
   }
 }
 
 export interface PlayerState {
-  resources: {
-    money: {
-      balance: number,
-      revenue: number
+  readonly resources: {
+    readonly money: {
+      readonly balance: number,
+      readonly revenue: number
     },
-    workers: {
-      max: number,
-      free: number
+    readonly workers: {
+      readonly max: number,
+      readonly free: number
     }
   },
-  victory: {
-    happiness: number,
-    goal: number
+  readonly victory: {
+    readonly happiness: number,
+    readonly goal: number
```

Comment:
we can fix this in a later PR, but I think `happiness` should have the same structure as `money` (with `balance` and `revenue` fields).

I think we should get rid of victory entirely, too....I am not sure we need it

Reply:
Did we ever decide to make happiness a cumulative resource?

Reply:
we (Alain and I?) have had some hallway discussions to that effect

in any case, `victory` feels weird - what is `goal`? I think we should just use a function to determine if victory has been achieved